tests/test_hashtable.py

```diff
@@ -243,63 +245,71 @@ def data_version_3_hash(self, data: np.ndarray):
                     group["values"] = np.array([b"ab", b"c", b"d"], dtype=object)
 
 
-def test_verify_chunk_reuse_nan(tmp_path):
+def test_chunk_reuse_nan(vfile):
     """Check that chunks are correctly verified when reused, even with nans."""
     data = np.array([1, 2, 3, 4, 5, np.nan])
-    filename = tmp_path / "testdata.h5"
-    with h5py.File(filename, mode="w") as f:
-        vf = VersionedHDF5File(f)
-        with vf.stage_version("r0") as sv:
-            sv.create_dataset("values", data=data, chunks=(6,))
-
-    with h5py.File(filename, mode="r+") as f:
-        vf = VersionedHDF5File(f)
-        with vf.stage_version("r1") as sv:
-            sv["values"] = np.concatenate((data, data))
+    with vfile.stage_version("r0") as sv:
+        sv.create_dataset("values", data=data, chunks=(6,))
+    with vfile.stage_version("r1") as sv:
+        sv["values"] = np.concatenate((data, data))
+    with vfile.stage_version("r2") as sv:
+        assert_slab_offsets(sv, "values", [0, 0])
```

Comment:
Before, there was no check that the chunks were actually reused!

Reply:
Good catch!